File: leetcode-submissions/214_Shortest_Palindrome.cpp
class Solution {
public:
    string shortestPalindrome(string s) {
        int i=0;
        for(int j=s.size()-1;j>=0 ;j--){
            if(s[i]==s[j]){
                i++;
            }
        }
        if(i==s.size()){
            return s;
        }
        //cecaaa
        string t=s.substr(i,s.size());
        reverse(t.begin(),t.end());
        //aaaec + "aa"+"cecaaa"
        return t+shortestPalindrome(s.substr(0,i))+s.substr(i);
    }
};

File: leetcode-submissions/214_Shortest_Palindrome.ts
function shortestPalindrome(s: string): string {
    let i:number=0;
    for(let j=s.length-1;j>=0;j--){
        if(s.charAt(i)===s.charAt(j)){
            i++;
        }
    }
    if(i===s.length){
        return s;
    }
    let p:string=s.split("").slice(i).reverse().join("");
    return p+shortestPalindrome(s.split("").slice(0,i).join(""))+s.split("").slice(i).join("");

};